src/main/java/org/testng/ITestNGMethod.java

```diff
@@ -154,10 +154,30 @@
 
   ITestNGMethod clone();
 
+  /**
+   * @deprecated - This method stands deprecated as of TestNG 7.0.0.
+   * Please use {@link #getRetryAnalyzer(ITestResult)} instead.
+   */
+  @Deprecated
   IRetryAnalyzer getRetryAnalyzer();
 
+  /**
+   * @deprecated - This method stands deprecated as of TestNG 7.0.0.
+   * Please use {@link #setRetryAnalyzerClass(Class)} instead.
+   */
+  @Deprecated
   void setRetryAnalyzer(IRetryAnalyzer retryAnalyzer);
 
+  default IRetryAnalyzer getRetryAnalyzer(ITestResult result) {
+    return null;
+  }
+
+  default void setRetryAnalyzerClass(Class<? extends IRetryAnalyzer> clazz) {}

```

Comment:
Here you must call `setRetryAnalyzer(....newInstance(clazz))`

Reply:
fixed